generators/csharp/model/src/union/UnionGenerator.ts

```diff
@@ -666,25 +665,24 @@ export class UnionGenerator extends FileGenerator<CSharpFile, ModelCustomConfigS
                                 default:
                                     assertNever(type.shape);
                             }
-                            writer.write(".Deserialize<");
-                            writer.writeNode(csharpType);
-                            writer.write(">(options)");
-                            if (csharpType.isOptional()) {
-                                writer.writeLine(",");
+                            if (csharpType.isReferenceType() === false) {
+                                // non-reference types can be always be deserialized directly as is
+                                writer.write(".Deserialize<", csharpType, ">(options)");
```

Comment:
Was the `writer.write` updated to support multiple args, and to write a node instead of using `writeNode`?

Reply:
Yes